src/frequenz/sdk/actor/_decorator.py

```diff
@@ -198,17 +198,23 @@ async def _start_actor(self) -> None:
 
             Raises:
                 asyncio.CancelledError: when the actor's task gets cancelled.
+
+            # noqa: DAR401 cancelled_error
             """
             logger.debug("Starting actor: %s", cls.__name__)
             number_of_restarts = 0
+            cancelled_error = None
             while True:
                 try:
                     await super().run()
-                except asyncio.CancelledError:
+                except asyncio.CancelledError as ex:
+                    cancelled_error = ex
                     logger.debug("Cancelling actor: %s", cls.__name__)
-                    raise
                 except Exception:  # pylint: disable=broad-except
                     logger.exception("Actor (%s) crashed", cls.__name__)
+                finally:
+                    if cancelled_error:
+                        raise cancelled_error
```

Comment:
I don't understand how moving `raise cancelled_error` from `except asyncio.CancelledError as ex:` to `finally:` helped. :D Could you please explain it ?

Reply:
Discussed. Thanks a lot.